src/app/_components/Event/EventSection.js

```diff
@@ -0,0 +1,154 @@
+      <Search
+        key={`searchicon+${index}`}
+        className="hidden h-4 w-4 transition-all group-hover:block group-focus:block"
+      />
+    );
+  }
+
+  // Fetch data and pass url params
+  const { data, error, isLoading, hasNextPage, fetchNextPage, isSuccess, isFetchingNextPage } = useInfiniteQuery({
+    queryFn: ({ month = activeMonth, pageParam = '' }) => allEvents({ month, take: 6, lastCursor: pageParam }),
+    queryKey: ['event', activeMonth],
+    getNextPageParam: lastPage => lastPage?.metaData.lastCursor,
+  });
+
+  // Get data on first render or on url open
+  useEffect(() => {
+    const monthFromQuery = searchParams.get('month');
+
+    if (monthFromQuery && monthFromQuery !== currentMonth) {
+      // If the month is defined in the query and is not equal to the current month
+      setActiveMonth(monthFromQuery);
+      removeParam();
+      addParam(monthFromQuery);
+    } else if (!monthFromQuery || monthFromQuery === undefined) {
+      setActiveMonth(currentMonth);
+      removeParam();
+      addParam(currentMonth);
+    }
+    // eslint-disable-next-line
+  }, [currentMonth]);
+
+  useEffect(() => {
+    // if the last element is in view and there is a next page, fetch the next page
+    if (inView && hasNextPage) {
+      fetchNextPage();
+    }
+    // eslint-disable-next-line
+  }, [hasNextPage, inView]);
+
+  // Filter data based on selected month
+  const handleFilter = newMonth => {
+    setActiveMonth(newMonth);
+    removeParam();
+    addParam(newMonth.toString());
+  };
+
+  return (
+    <>
+      <div className="lg:w-max-[900px] mx-auto flex w-full flex-col items-start justify-start gap-6 self-stretch">
+        <div className="flex flex-row flex-wrap items-start justify-start gap-3">
+          {filteredMonths.map(month => (
+            <PillButton
+              variant="eventFilter"
+              colorVariant="semiorange"
+              className={cn(activeMonth - 1 === month.index && activeButtonStyles, 'group w-fit')}
+              key={month.index}
+              onClick={() => {
+                handleFilter(month.index + 1);
+              }}
+              icon={selectButtonIcon(month.index)}
+            >
+              {month.name.charAt(0).toUpperCase() + month.name.slice(1)}
+            </PillButton>
+          ))}
+        </div>
+
+        <ul className="grid w-full gap-4 self-stretch sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3">
+          {isSuccess &&
+            data?.pages.map(page =>
+              page.data.map((event, index) => {
+                if (page.data.length === index + 1)
+                  return (
+                    <div ref={ref} key={event.id}>
+                      <EventCard event={event} />
+                    </div>
+                  );
+
+                return <EventCard key={event.id} event={event} />;
+              }),
+            )}
+        </ul>
+      </div>
+
+      {(isLoading || isFetchingNextPage) && <CircularProgress />}
```

Comment:
@Yivgha can we use skeleton here? Page is jumping every time when I’m changing month in filter for the first time.

Reply:
created Skeletoncard

Reply:
@Yivgha I guess we don't need Progress in the same time with Skeleton then.

Reply:
deleted Progress